homeassistant/components/media_player/sonos.py

```diff
@@ -834,6 +858,22 @@ def media_title(self):
 
         return self._media_title
 
+    @property
+    def night_sound(self):
+        """Get status of Night Sound."""
+        if self._coordinator:
```

Comment:
You defer to the coordinator. Are these options really per-group (like a playlist), not per-speaker (like volume)?

Reply:
I did some research using two speakers (Kitchen, Den) of my multi-unit setup. Den is a Playbar that supports the night_sound (NS) and speech_enhance (SE) features when playing from the TV source.

1. When I build a group designating Kitchen as the coordinator and then switch to the TV source the Den then becomes the coordinator.
2. When playing TV in a group the NS and SE effects are audible on the Den but not the Kitchen.
3. SoCo will not send NS and SE commands to a non-Playbar speaker.

Based on #2 I agree with you that this is a per-speaker feature. This was masked by the Sonos behavior described in #1.

I'll push a new version that does not defer to the coordinator momentarily. It will not include a check to ensure that the speaker is the coordinator but given #3 this should not cause an error.

Sound good? Thank you for the close review.

Reply:
I pushed an updated version per your comments. I also rebased to the current dev branch.

Reply:
I like your changes though if the Playbar always ends up being the coordinator, I guess it doesn't matter much.